modules/fluid_properties/src/userobjects/Water97FluidProperties.C

```diff
@@ -0,0 +1,1411 @@
+    std::pow(rhobar + 2.392190, 2.0));
+
+  // Critical enhancement
+  Real DeltaT = std::abs(Tbar - 1.0) + 0.00308976;
+  Real Q = 2.0 + 0.0822994 / std::pow(DeltaT, 0.6);
+  Real S = (Tbar >= 1.0 ? 1.0 / DeltaT : 10.0932 / std::pow(DeltaT, 0.6));
+
+  Real lambda2 = (0.0701309 / std::pow(Tbar, 10.0) + 0.011852) * std::pow(rhobar, 1.8) *
+    std::exp(0.642857 * (1.0 - std::pow(rhobar, 2.8))) + 0.00169937 * S *
+    std::pow(rhobar, Q) * std::exp((Q / (1.0 + Q)) * (1.0 - std::pow(rhobar, 1.0 + Q))) -
+    1.02 * std::exp(-4.11717 * std::pow(Tbar, 1.5) - 6.17937 / std::pow(rhobar, 5.0));
+
+  return lambda0 + lambda1 + lambda2;
+}
+
+Real
+Water97FluidProperties::s(Real pressure, Real temperature) const
+{
+  Real entropy, pi, tau, density3, delta;
+
+   // Determine which region the point is in
+   unsigned int region = inRegion(pressure, temperature);
+   switch (region)
+   {
+     case 1:
+       pi = pressure / _p_star[0];
+       tau = _T_star[0] / temperature;
+       entropy = _Rw * (tau * dgamma1_dtau(pi, tau) - gamma1(pi, tau));
+       break;
+
+     case 2:
+       pi = pressure / _p_star[1];
+       tau = _T_star[1] / temperature;
+       entropy = _Rw * (tau * dgamma2_dtau(pi, tau) - gamma2(pi, tau));
+       break;
+
+     case 3:
+       // Calculate density first, then use that in Helmholtz free energy
+       density3 = densityRegion3(pressure, temperature);
+       delta = density3 / _rho_critical;
+       tau = _T_star[2] / temperature;
+       entropy = _Rw * (tau *  dphi3_dtau(delta, tau) - phi3(delta, tau));
+       break;
+
+     case 5:
+       pi = pressure / _p_star[4];
+       tau = _T_star[4] / temperature;
+       entropy = _Rw * (tau * dgamma5_dtau(pi, tau) - gamma5(pi, tau));
+       break;
+    }
+  // Output in kJ/kg
+  return entropy / 1000.0;
+}
+
+Real
+Water97FluidProperties::h(Real pressure, Real temperature) const
+{
+  Real enthalpy, pi, tau, delta;
+
+   // Determine which region the point is in
+   unsigned int region = inRegion(pressure, temperature);
+   switch (region)
+   {
+     case 1:
+       pi = pressure / _p_star[0];
+       tau = _T_star[0] / temperature;
+       enthalpy = _Rw * _T_star[0] * dgamma1_dtau(pi, tau);
+       break;
+
+     case 2:
+       pi = pressure / _p_star[1];
+       tau = _T_star[1] / temperature;
+       enthalpy = _Rw * _T_star[1] * dgamma2_dtau(pi, tau);
+       break;
+
+     case 3:
+     {
+       // Calculate density first, then use that in Helmholtz free energy
+       Real density3 = densityRegion3(pressure, temperature);
+       delta = density3 / _rho_critical;
+       tau = _T_star[2] / temperature;
+       enthalpy = _Rw * temperature * (tau *  dphi3_dtau(delta, tau) +
+         delta *  dphi3_ddelta(delta, tau));
+       break;
+     }
+
+     case 5:
+       pi = pressure / _p_star[4];
+       tau = _T_star[4] / temperature;
+       enthalpy = _Rw * _T_star[4] * dgamma5_dtau(pi, tau);
+       break;
+    }
```

Comment:
This block should be indented by 2 spaces (3 here) and the closing `}` needs to match the indentation of the switch, i.e. 2 spaces